### PR TITLE
Filtered archived repositories when discovering repos

### LINKED ISSUE #5239: Archived Repositories are not removed from Backstage
I'm using `github-discovery` to scan and import repositories in my organization to Backstage. When a repository is archived, it still remains in Backstage (and there's no way to remove it).

## Expected Behavior

Archived repos should be removed from view.

## Current Behavior

Archived repos are forever left in the UI and the DB.

## Steps to Reproduce

1. Use `github-discovery` to scan an entire org's repos.
2. Archive one of the repos that was added to Backstage.
3. Repo still appear in Backstage.

## Context

`github-discovery` is the main method we use for syncing Backstage with our repositories.

## Your Environment

- NodeJS Version (v12): v14
- Operating System and Version (e.g. Ubuntu 14.04): Mac OS X
- Browser Information: Chrome

### PR DESCRIPTION
**Pull request by nirga**
## Hey, I just made a Pull Request!

Fixes #5239. Filtering of archived repositories so they won't be added to Backstage.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/real-apples-visit.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-catalog-backend': patch
+---
+
+Filtering of archived repositories so they won't be added to Backstage.
```
Comment (timbonicus) [inline .changeset/real-apples-visit.md:5]:
```suggestion
GithubDiscoveryProcessor now excludes archived repositories so they won't be added to Backstage.
```

File: plugins/catalog-backend/src/ingestion/processors/GithubDiscoveryProcessor.ts [hunk 2]
```diff
@@ -78,7 +78,9 @@ export class GithubDiscoveryProcessor implements CatalogProcessor {
     this.logger.info(`Reading GitHub repositories from ${location.target}`);
 
     const { repositories } = await getOrganizationRepositories(client, org);
-    const matching = repositories.filter(r => repoSearchPath.test(r.name));
+    const matching = repositories.filter(
+      r => r.isArchived || repoSearchPath.test(r.name),
```
Comment (timbonicus) [inline plugins/catalog-backend/src/ingestion/processors/GithubDiscoveryProcessor.ts:82]: Hmm, I think the logic is backwards here?

```
const matching = repositories.filter(r => !r.isArchived && repoSearchPath.test(r.name));
```

I'm wondering how the test passed, I think you might need to make sure it's only called with the resulting location once:

```
expect(emitter).toHaveBeenCalledTimes(1);
```

Unless I'm missing something?

**Review by timbonicus — COMMENTED** [review]
Super awesome contribution! 🙌

**Review by freben — APPROVED** [review]
nice!

**Pull request merged**